x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/autoscaling/MlAutoscalingResourceTracker.java

```diff
@@ -208,47 +219,118 @@ static void getMemoryAndProcessors(
                 extraModelMemoryInBytes += estimatedMemoryUsage;
 
-                // if not low priority, check processor requirements
+                // if not low priority, check processor requirements. If
                 if (Priority.LOW.equals(modelAssignment.getValue().getTaskParams().getPriority()) == false) {
-                    // as assignments can be placed on different nodes, we only need numberOfThreadsPerAllocation here
-                    extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
-                    extraProcessors += numberOfAllocations * numberOfThreadsPerAllocation;
+                    if (numProcessorsNeeded > numberOfAvailableProcessors) {
+                        // as assignments can be placed on different nodes, we only need numberOfThreadsPerAllocation here
+                        extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
+                        extraProcessors += numberOfRequestedAllocations * numberOfThreadsPerAllocation - numExistingProcessorsToBeUsed;
+                    }
+                    processorsNewlyAssignedToTrainedModelsAllocationsThisTime += numExistingProcessorsToBeUsed;
                 }
             } else if (assignment.getNodeRoutingTable().values().stream().allMatch(r -> r.getState().consumesMemory() == false)) {
-                // Ignore states that don't consume memory, for example all allocations are failed
+                // Ignore states that don't consume memory, for example all allocations are failed or stopped
                 continue;
+            } else if (AssignmentState.STARTED.equals(assignment.getAssignmentState()) && numMissingAllocations > 0) {
+                // this state occurs when we have a partial allocation, but not fully allocated
+
+                long numMemoryBytesNeeded = assignment.getTaskParams().estimateMemoryUsageBytes();
+
+                if (numberOfAvailableProcessors < numProcessorsNeeded || perNodeAvailableModelMemoryBytes < numMemoryBytesNeeded) {
+                    // we don't have enough processors or memory to start the new allocations, we need to scale up
+
+                    processorsNewlyAssignedToTrainedModelsAllocationsThisTime += numExistingProcessorsToBeUsed;
+
+                    extraProcessors += numMissingAllocations * numberOfThreadsPerAllocation - numExistingProcessorsToBeUsed;
+                    extraSingleNodeProcessors = Math.max(extraSingleNodeProcessors, numberOfThreadsPerAllocation);
+                    extraModelMemoryInBytes += numMemoryBytesNeeded;
+                    extraPerNodeModelMemoryBytes = Math.max(extraPerNodeModelMemoryBytes, numMemoryBytesNeeded);
+                    final int extraProcessorsNeededForAssignment = numMissingAllocations * numberOfThreadsPerAllocation
+                        - numExistingProcessorsToBeUsed;
+                    logger.warn(
```

Comment:
Should this be a log info?

Reply:
It could be. I thought warn made sense since this is useful/important information about the state of the system. Maybe warn is a bit extreme though?